paddle/fluid/operators/affine_channel_op.cc

```diff
@@ -295,20 +295,20 @@ class AffineChannelNoNeedBufferVarsInference
   using framework::NoNeedBufferVarsInference::NoNeedBufferVarsInference;
 
  private:
-  inline bool HasInput(const std::string& name) const {
-    auto& inputs = Inputs();
-    auto iter = inputs.find(name);
-    if (iter == inputs.end() || iter->second.empty()) {
+  inline bool HasOutput(const std::string& name) const {
+    auto& outputs = Outputs();
+    auto iter = outputs.find(name);
+    if (iter == outputs.end() || iter->second.empty()) {
       return false;
     } else {
       return iter->second[0] != framework::kEmptyVarName;
     }
   }
 
  public:
-  std::unordered_set<std::string> operator()() const {
-    if (!HasInput(framework::GradVarName("Scale")) &&
-        !HasInput(framework::GradVarName("Bias"))) {
+  std::unordered_set<std::string> operator()() const override {
+    if (!HasOutput(framework::GradVarName("Scale")) &&
+        !HasOutput(framework::GradVarName("Bias"))) {
```

Comment:
Since, Out = Scale*X + Bias, do we have to have X's data when computing  Bias@Grad?

Can we write:

if (!HasOutput(framework::GradVarName("Scale"))) {
    return {"X"};
}

Reply:
Same comment as develop branch.

Reply:
Discussed offline, the OP calculates Scale@Grad and Bias@Grad at the same time. In other words, the two Grads must occur at the same time. The PR LGTM